library/general/src/lib/cfa/nsswitch.rb

```diff
@@ -0,0 +1,273 @@
+  #   file = Nsswitch.new
+  #   file.load
+  #   file.services_for("hosts") #=> ["db", "files"]
+  #
+  # @example Writing service specifications for a database
+  #   file = Nsswitch.new
+  #   file.load
+  #   file.update_entry("hosts", ["db", "files"])
+  #   file.save
+  #
+  # @example Deleting a database entry
+  #   file = Nsswitch.new
+  #   file.load
+  #   file.delete_entry("hosts")
+  #   file.save
+  #
+  # @example Loading shortcut
+  #   file = Nsswitch.load
+  #   file.entries #=> ["passwd", "group", "shadow", "hosts", "networks", "aliases"]
+  class Nsswitch < BaseModel
+    include Yast::Logger
+
+    # Path to local configuration file
+    LOCAL_PATH = "/etc/nsswitch.conf".freeze
+    private_constant :LOCAL_PATH
+
+    # Path to vendor configuration file
+    VENDOR_PATH = "/usr/etc/nsswitch.conf".freeze
+    private_constant :VENDOR_PATH
+
+    class << self
+      # Instantiates and loads the file
+      #
+      # This method is basically a shortcut to instantiate and load the content in just one call.
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Nsswitch] File with the already loaded content
+      def load(file_handler: Yast::TargetFile)
+        new(file_handler: file_handler).tap(&:load)
+      end
+    end
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_handler: Yast::TargetFile)
+      super(AugeasParser.new("nsswitch.lns"), load_path, file_handler: file_handler)
+    end
+
+    # Loads the file content
+    #
+    # If the file does not exist, consider it as empty.
+    #
+    # @see CFA::BaseModel#load
+    def load
+      super
+      fix_collection_names(data)
+      @current_content = @parser.serialize(data)
+    rescue Errno::ENOENT # PATH does not exist yet
+      self.data = @parser.empty
+      @loaded = true
+    end
+
+    # Returns a list of defined databases
+    #
+    # @return [Array<String>] List of database names
+    def entries
+      databases.keys
+    end
+
+    # Service specifications for given database name
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    #
+    # @return [Array<String>] database service specifications
+    def services_for(db_name)
+      databases[db_name]&.services
+    end
+
+    # Update (or create if it does not exist yet) the entry for given database name
+    #
+    #
+    # @note Current services, if any, are completely replaced
+    #
+    # @see #entry_for
+    # @see DatabaseEntry#services=
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    # @param services [Array<String>] service specifications for the database
+    def update_entry(db_name, services)
+      database = entry_for(db_name)
+      database.services = services
+    end
```

Comment:
### :warning:  note that actions (a.k.a. reactions) are not supported. :warning: 

This could be problematic limitation since `#update_entry` has been designed to fully replace the service specifications for a database **without** taking care about actions. Then, previous configuration contains 

  ```
   hosts:          dns [!UNAVAIL=return] files
  ```

  and a `CFA::Nsswitch#update_entry("hosts", ["db", "dns", "files"])` is performed, the result will be a file containing 

  ```
   hosts:          db dns files
  ```

This is not a problem with the current [yast-pam/Nsswitch](https://github.com/yast/yast-pam/blob/96a24a906231730c90bca79d3754da9cc4e70b1c/src/modules/Nsswitch.rb) module implementation because it _just replace a fragment of a line in a text file_ with the given value, usually [the result of manipulate the previous read value](https://github.com/search?q=%22Nsswitch.WriteDb%22+user%3Ayast&type=Code).

Adding `#append_service_to` and `#prepend_service_to` **does not** help because the problem still being the same, actions are not supported and the code will fully replace the service specifications with the new ones.

Do you think it worth it to add support for such complex structure at this time? (I think so, but I'd like to know others POV before proceeding)

Reply:
well, idea of CFA is to not modify what is not needed or what we do not understand. So CFA tries quite heavy to keep existing stuff ( including comments, etc. ). So I am not sure about support, but we should at least not break it.

Reply:
Well, thinking again about it, I see some similarities with other code we have. I think it is ok to remove actions as long as there is warning in UI that it will completely replace entries.